linera-core/src/client.rs

```diff
@@ -746,9 +760,28 @@ where
                 .flatten()
                 .collect::<Vec<_>>();
                 if !values.is_empty() {
-                    self.process_certificate(certificate.clone(), values)
+                    self.process_certificate(certificate.clone(), values, BTreeMap::new())
                         .await?;
                 }
+            } else if let LocalNodeError::WorkerError(WorkerError::BlobsNotFound(blob_ids)) = &err {
+                let blobs = future::join_all(blob_ids.iter().map(|blob_id| {
+                    LocalNodeClient::<S>::download_blob(nodes.clone(), block.chain_id, *blob_id)
+                }))
+                .await
+                .into_iter()
+                .flatten()
+                .collect::<Vec<_>>();
+                if !blobs.is_empty() {
```

Comment:
That being empty should be unreachable, I think? We could `ensure!` it, or just not have special handling for that case.